### PR TITLE
tools: move a location of formatter, make a script of c++, add README

### PR DESCRIPTION
**Pull request by sunghan-chang**
1. Move existing c formatter from os/tools to tools.
  Formatter is not only for os. Tools folder is approprietary location.
2. Add new script to run c++ formatter which use clang-format
  To give same usages with c, make a new script which has same usages.
3. Add new README to give information of formatter
4. Check existance of necessary package on ubuntu before excuting

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [712dbab89d1a28759a2524a4e30d233aea22f2aa] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [712dbab89d1a28759a2524a4e30d233aea22f2aa] - Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@dr-venkman @lokeshbv I am so sorry but could you help my English sentence?

**Comment by seinfra** [comment]
Target : [682d068af88879d30ce101afc350d98bc60acf35] -

**Comment by seinfra** [comment]
Target : [682d068af88879d30ce101afc350d98bc60acf35] -

**Comment by seinfra** [comment]
Target : [e229399172e85caa7f33ab59f0fa1e525c7f4d5f] - Code Rule Check (C++) OK.

File: tools/HowToUseFormatter.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+usage: ./tools/cformatter.sh TARGET
+     TARGET: folder path or target files
+Apply TizenRT C coding rule
+
+example:
+    ./tools/cformatter.sh ../kernel
+    ./tools/cformatter.sh ../../abc.c
+```
+Note! This sometimes makes incorrect changes.  
+Ususally it is a space between type casting and variable as show below:
```
Comment (lokeshbv) [inline tools/HowToUseFormatter.md]: Ususally ---> Usually
 "Usually it adds a space between type casting and variable as show below"

File: tools/HowToUseFormatter.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# How to use formatter
+Here are two formatter (or beautifuler) for C and C++ languages, cformatter.sh and cxxformatter.sh.
```
Comment (lokeshbv) [inline tools/HowToUseFormatter.md]: beautifuler --> Beautifier

File: tools/HowToUseFormatter.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+## Prerequisites
+```bash
+sudo apt-get install indent astyle
+sudo apt-get install clang-format
+```
+
+## formatter for C
+The **cformatter** script applies TizenRT C coding rule.  
+```--help``` shows how to use it.
```
Comment (lokeshbv) [inline tools/HowToUseFormatter.md]: > --help option shows how to use it

File: tools/HowToUseFormatter.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+Note! This sometimes makes incorrect changes.  
+Ususally it is a space between type casting and variable as show below:
+```
+buf = (int *) malloc(8);
+size_a = (uint8_t) size_b;
+```
+
+## formatter for C++
+The **cxxformatter** script applies TizenRT C++ coding rule.  
+As C does, please type ```--help``` to know the usages.
```
Comment (lokeshbv) [inline tools/HowToUseFormatter.md]: > Type cxxformatter --help to know the usage

File: tools/HowToUseFormatter.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+@ubuntu:~/TizenRT$ ./tools/cformatter.sh --help
+usage: ./tools/cformatter.sh TARGET
+     TARGET: folder path or target files
+Apply TizenRT C coding rule
+
+example:
+    ./tools/cformatter.sh ../kernel
+    ./tools/cformatter.sh ../../abc.c
+```
+Note! This sometimes makes incorrect changes.
```
Comment (lokeshbv) [inline tools/HowToUseFormatter.md]: Note! ---> **Note:**

**Comment by seinfra** [comment]
Target : [6698c790534081a9d99f834a64e57b16b66d444d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6698c790534081a9d99f834a64e57b16b66d444d] - Code Rule Check OK.

**Pull request merged**